### PR TITLE
Update handling of reaction roles

### LINKED ISSUE #511: Separate internal standards from products
Listing internal standards as a product (e.g., for the sake of reporting raw UV areas) might lead to unusual reaction SMILES strings. We could separate out internal standard measurements in a separate `repeated ReactionProduct standards` field. I'm not completely sure if this is the best way to structure it.

### PR DESCRIPTION
**Pull request by michaelmaser**
Fixes #511. 

Along with the proto changes, updated notebooks, message helpers, etc. Seems to be working in those places. Creating as a draft PR in case it breaks anything else I missed. Let me know what you think!

### CODE REVIEW

File: ord_schema/message_helpers.py
Comment (skearnes) [inline ord_schema/message_helpers.py:374]: nit; the enum values are class attributes, not instance attributes; so use `roles = reaction_pb2.ReactionRole` here instead of creating an instance of `ReactionRole`.

File: ord_schema/proto/reaction.proto
Comment (skearnes) [inline ord_schema/proto/reaction.proto:872]: Please add a comment e.g. must be one of PRODUCT or INTERNAL_STANDARD

File: ord_schema/proto/reaction.proto
Comment (skearnes) [inline ord_schema/proto/reaction.proto:839]: Just `Product`? Why the change?

Comment (michaelmaser) [reply ord_schema/proto/reaction.proto:839]: Request from @connorcoley in #511:

>In this same change, I'd suggest we rename ReactionProduct to ProductCompound

File: ord_schema/proto/reaction.proto
Comment (michaelmaser) [reply ord_schema/proto/reaction.proto:872]: I thought we were allowing any reaction role to be specified. That way, e.g., recovered starting materials could be specified with `reaction_role` `REACTANT` even though they're `ProductCompound`s. Can add a comment to help clarify this, if what I'm thinking is correct.

File: ord_schema/message_helpers.py
Comment (michaelmaser) [reply ord_schema/message_helpers.py]: Fixed. Thanks!

File: ord_schema/proto/reaction.proto
Comment (skearnes) [reply ord_schema/proto/reaction.proto:872]: Ah, that's right; a comment would be great.

Comment (michaelmaser) [reply ord_schema/proto/reaction.proto:877]: Done, thanks!

**Comment by skearnes** [comment]
PS changed `Addresses` to `Fixes` in the PR description; see https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Review by skearnes — APPROVED** [review]
Please wait for approval from @connorcoley before merging.

**Comment by michaelmaser** [comment]
>PS changed `Addresses` to `Fixes` in the PR description...

Ah, got it, thanks!

>Please wait for approval from @connorcoley before merging.

Will do.

File: ord_schema/message_helpers_test.py
Comment (connorcoley) [inline ord_schema/message_helpers_test.py:136]: For future reference, you can write `reaction_role='PRODUCT'` instead of `reaction_role=reaction_pb2.ReactionRole.PRODUCT`

File: ord_schema/validations.py
Comment (connorcoley) [inline ord_schema/validations.py:1071]: If I'm allowed to be picky, can you rename `validate_reaction_product` to `validate_product_compound`?

File: ord_schema/message_helpers_test.py
Comment (connorcoley) [inline ord_schema/message_helpers_test.py:154]: Same here

**Review by connorcoley — APPROVED** [review]
Two very minor changes suggested (renaming validation function & using shorter method of defining new product roles in the message helpers test), otherwise looks good!

File: ord_schema/message_helpers_test.py
Comment (michaelmaser) [reply ord_schema/message_helpers_test.py]: Great, thanks. Wasn't sure. Saw this construct elsewhere and used it here to be safe. Updated!

File: ord_schema/validations.py
Comment (michaelmaser) [reply ord_schema/validations.py]: Absolutely! Fixed.

File: ord_schema/message_helpers_test.py
Comment (michaelmaser) [reply ord_schema/message_helpers_test.py]: Done, thanks!

**Comment by michaelmaser** [comment]
Thanks for the reviews!

**Pull request merged**
